src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -11516,31 +11572,53 @@ public LocalState Clone()
             public bool Join(in LocalState other)
             {
                 Debug.Assert(Id == other.Id);
-                bool result = false;
+                bool hasChanged = false;
                 if (_container is { } && _container.Value.Join(in other._container!.Value))
                 {
-                    result = true;
+                    hasChanged = true;
                 }
-                if (_state.UnionWith(in other._state))
+
+                Debug.Assert(_state.Capacity == other._state.Capacity);
+                var oldReachable = Reachable;
```

Comment:
Could you help me understand why the old "union" and "intersection" operations aren't the right thing to do any more?

Reply:
"union" and "intersection" are BitVector concepts. They worked because of the values we were using false-false, false-true and true-true. But they don't work with the new values.

Reply:
It sounds like our states used to have a property where "flipping a given bit to 1" strictly put us in a "more pessimistic" and still coherent state. And similarly, "flipping a given bit to 0" strictly put us in a "more optimistic" and still coherent state. And we no longer have that property. Is that about right?

Is there any risk of perf being reduced in this approach? It seems like Join is done a lot, so there's a risk that an operation that used to be "vectorized" in a sense is now being done by many "serial" bitwise and bit-shifting operations instead.